core/src/com/unciv/logic/civilization/PolicyManager.kt

```diff
@@ -24,8 +24,15 @@ class PolicyManager {
     var numberOfAdoptedPolicies = 0
     var shouldOpenPolicyPicker = false
         get() = field && canAdoptPolicy()
-    var legalismState = HashMap<String, String>()
+
+    private var cultureBuildingsAdded = HashMap<String, String>() // Maps cities to buildings
+    private var specificBuildingsAdded = HashMap<String, MutableSet<String>>() // Maps buildings to cities
+    private var maintenanceFreeBuildings = HashMap<String, MutableSet<String>>() // Maps cities to buildings
```

Comment:
I fail to see why you need this - isn't the information you need already present in the above 2 new variables? We don't need to save this as part of the state

Reply:
Was mostly for caching for easy access, but you could indeed derive this information from the above maps. I'll remove it.